freshdesk/v2/test.py

```diff
@@ -124,15 +124,59 @@ def test_api_prefix(self):
         self.assertEqual(api._api_prefix,
                          'https://test_domain.freshdesk.com/api/v2/')
 
+    @responses.activate
+    def test_400_error(self):
+        responses.add(responses.GET,
+                      'https://{}/api/v2/tickets/1'.format(DOMAIN),
+                      status=400)
+
+        api = API('pythonfreshdesk.freshdesk.com', 'test_key')
+        from freshdesk.v2.errors import FreshdeskBadRequest
+        with self.assertRaises(FreshdeskBadRequest):
+            api.tickets.get_ticket(1)
+
     @responses.activate
     def test_403_error(self):
         responses.add(responses.GET,
                       'https://{}/api/v2/tickets/1'.format(DOMAIN),
                       status=403)
 
         api = API('pythonfreshdesk.freshdesk.com', 'invalid_api_key')
-        from requests.exceptions import HTTPError
-        with self.assertRaises(HTTPError):
+        from freshdesk.v2.errors import FreshdeskAccessDenied
+        with self.assertRaises(FreshdeskAccessDenied):
+            api.tickets.get_ticket(1)
+
+    @responses.activate
+    def test_404_error(self):
+        responses.add(responses.GET,
+                      'https://{}/api/v2/tickets/1'.format(DOMAIN),
+                      status=404)
+
+        api = API('pythonfreshdesk.freshdesk.com', 'test_key')
+        from freshdesk.v2.errors import FreshdeskNotFound
+        with self.assertRaises(FreshdeskNotFound):
+            api.tickets.get_ticket(1)
+
+    @responses.activate
+    def test_rate_limited_error(self):
+        responses.add(responses.GET,
+                      'https://{}/api/v2/tickets/1'.format(DOMAIN),
+                      status=429)
+
+        api = API('pythonfreshdesk.freshdesk.com', 'test_key')
+        from freshdesk.v2.errors import FreshdeskRateLimited
+        with self.assertRaises(FreshdeskRateLimited):
+            api.tickets.get_ticket(1)
+
+    @responses.activate
+    def test_50x_error(self):
+        responses.add(responses.GET,
+                      'https://{}/api/v2/tickets/1'.format(DOMAIN),
+                      status=502)
+
+        api = API('pythonfreshdesk.freshdesk.com', 'test_key')
+        from freshdesk.v2.errors import FreshdeskServerError
+        with self.assertRaises(FreshdeskServerError):
```

Comment:
These new tests could've been rolled into one, but I wasn't sure it was worth adding a new [dependency](https://pypi.org/project/parameterized/) for that.

Reply:
I wonder if it's worth swapping to pytest? It's a drop-in replacement and you get [parametrize for free](https://docs.pytest.org/en/latest/parametrize.html)!

Reply:
That's a good idea! Maybe in a separate PR though, to avoid mixing things? I can refactor the tests then.

Reply:
Looks like in the case of `pytest` we would need to install an [additional package](https://pypi.org/project/pytest-cov) as well, except for measuring code coverage instead of parametrizing tests 😄 
Does it still make sense to switch?

Reply:
Yeah, that's OK. Just drop the needed packages into `test-requirements.txt` so they don't get pulled down with a `pip install`.